ComponentKit/Components/CKButtonComponent.mm

```diff
@@ -161,7 +161,7 @@ + (instancetype)newWithTitles:(const std::unordered_map<UIControlState, NSString
                             std::move(attributes),
                             {
                               .accessibilityLabel = accessibilityConfiguration.accessibilityLabel,
-                              .accessibilityComponentAction = enabled ? action : NULL
+                              .accessibilityComponentAction = enabled ? CKComponentAction(action) : NULL
```

Comment:
In this callsite, we're basically demoting the action, which shows the power of this approach. Before what was happening is we were passing a component action to a system that didn't provide *any* context object. This meant that at runtime, actions sent from this API would silently fail to provide any arguments, leaving you with nil events. This is a little unexpected.

I've provided an explicit constructor to allow conversion of these types to allow for demotion, but it does force you to be aware that it's happening.